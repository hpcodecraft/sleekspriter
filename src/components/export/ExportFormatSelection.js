import React from "react";
import { t } from "../../utils";

const ExportFormatSelection = ({ format, part, setFormat }) => {
  let formats = ["png", "jpg"];
  if(part === "animation") {
    formats = ["gif"]; //, "mov"];
  }

  return (
    <div>
      <h6>{t("Output")}</h6>
      <ul>
        <li>
          <select onChange={(e) => setFormat(e.target.value)} value={format}>
            {formats.map(function(format) {
              return( <option key={format} value={format === "jpg" ? "jpeg" : format}>{format}</option> );
            }, this)}
          </select>
        </li>
      </ul>
    </div>
  );
};

export default ExportFormatSelection;
