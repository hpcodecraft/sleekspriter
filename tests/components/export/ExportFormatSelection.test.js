import ExportFormatSelection from "components/export/ExportFormatSelection";

describe("ExportFormatSelection", () => {
  let props, wrapper;
  beforeEach(() => {
    props = {
      format: "png",
      part: "spritesheet",
      setFormat: jest.fn(),
    };

    wrapper = shallow(<ExportFormatSelection {...props} />);
  });

  it("should render correctly", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  describe("select field", () => {
    it("has gif as option when an animation is exported", () => {
      props.part = "animation";
      wrapper = shallow(<ExportFormatSelection {...props} />);
      expect(wrapper.find('option[value="gif"]').length).toEqual(1);
    });

    it("does not have gif as option when no animation is exported", () => {
      expect(wrapper.find('option[value="gif"]').length).toEqual(0);
    });

    it("fires setFormat handler on change", () => {
      wrapper.find("select").simulate("change", { target: { value: "jpg" } });
      expect(props.setFormat).toHaveBeenCalledWith("jpg");
    });
  });
});
