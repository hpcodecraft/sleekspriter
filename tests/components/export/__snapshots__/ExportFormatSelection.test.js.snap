// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExportFormatSelection should render correctly 1`] = `"<div><h6>Save as</h6><ul><li><select><option selected=\\"\\" value=\\"png\\">png</option><option value=\\"jpeg\\">jpg</option></select></li></ul></div>"`;
